packages/react-jss/src/utils/sheets.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+import warning from 'tiny-warning'
+import {getDynamicStyles, type StyleSheetFactoryOptions} from 'jss'
+import type {StyleSheet} from 'jss'
+import type {Context, DynamicRules, Styles} from '../types'
+import {getManager} from './managers'
+import {jss as defaultJss} from '../jss'
+import {addMetaForSheet, getMetaForSheet} from './sheets-meta'
+
+interface Options<Theme> {
+  context: Context;
+  theme: Theme;
+  name: string;
+  index: number;
+  styles: Styles<Theme>;
+  sheetOptions: $Diff<StyleSheetFactoryOptions, {index: number | void}>;
+}
+
+const getStyles = <Theme>(options: Options<Theme>) => {
+  const {styles} = options
+  if (typeof styles !== 'function') {
+    return styles
+  }
+
+  warning(
+    styles.length !== 0,
+    `[JSS] <${
+      options.name
+    } />'s styles function doesn't rely on the "theme" argument. We recommend declaring styles as an object instead.`
+  )
+
+  return styles(options.theme)
+}
+
+function getSheetOptions<Theme>(options: Options<Theme>, link: boolean) {
+  const classNamePrefix =
+    process.env.NODE_ENV === 'production' ? '' : `${options.name.replace(/\s/g, '-')}-`
+
+  return {
+    ...options.sheetOptions,
+    ...options.context.sheetOptions,
+    index: options.index,
+    meta: `${options.name}, ${typeof options.styles === 'function' ? 'Themed' : 'Unthemed'}`,
+    classNamePrefix: options.context.sheetOptions.classNamePrefix + classNamePrefix,
+    link
+  }
+}
+
+function createStaticSheet<Theme>(options: Options<Theme>) {
+  const manager = getManager(options.context, options.index)
+  const existingSheet = manager.get(options.theme)
+
+  if (existingSheet) {
+    return existingSheet
+  }
+
+  const jss = options.context.jss || defaultJss
+  const styles = getStyles(options)
+  const dynamicStyles = getDynamicStyles(styles)
+  const sheet = jss.createStyleSheet(styles, getSheetOptions(options, dynamicStyles !== null))
+
+  addMetaForSheet(sheet, {
+    dynamicStyles,
+    styles,
+    dynamicRuleCounter: 0
+  })
+
+  manager.add(options.theme, sheet)
+
+  return sheet
+}
+
+const removeDynamicRules = (sheet: ?StyleSheet, rules: ?DynamicRules) => {
```

Comment:
can StyleSheet be `null`? I would try to avoid maybe types as much as possible.

Reply:
Yes, it can be `undefined`